apps/cowswap-frontend/src/modules/bridge/updaters/PendingBridgeOrdersUpdater.tsx

```diff
@@ -39,20 +40,53 @@ function processExecutedBridging(crossChainOrder: CrossChainOrder): void {
 interface PendingOrderUpdaterProps {
   chainId: SupportedChainId
   orderUid: string
+  openSince?: number
 }
 
-function PendingOrderUpdater({ chainId, orderUid }: PendingOrderUpdaterProps): ReactNode {
+function PendingOrderUpdater({ chainId, orderUid, openSince }: PendingOrderUpdaterProps): ReactNode {
   const { data: crossChainOrder } = useCrossChainOrder(chainId, orderUid)
   const updateBridgeOrderQuote = useUpdateBridgeOrderQuote()
   const addOrderToSurplusQueue = useAddOrderToSurplusQueue()
   const analytics = useCowAnalytics()
+  const waitingTooLongNpsTriggeredRef = useRef(false)
+
+  // Check once a minute the time it has been pending and trigger appzi if greater than threshold
+  useEffect(() => {
```

Comment:
What if:
 - bridge order is open for 7 minutes
 - it triggers the Appzi survey
 - reload the page 
 
Will it trigger the survey again?

Reply:
No, because appzi controls the notification on their side and only do it once (the localStorage key).
This is just to prevent us hitting their API every minute needlessly.